src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -196,18 +203,28 @@ public static bool MonitoringIsEnabled
         public void SetCachePath(string path) { }
         public void SetShadowCopyFiles() { }
         public void SetShadowCopyPath(string path) { }
-        // TODO
-        public Assembly[] GetAssemblies() { throw null; }
-        public event AssemblyLoadEventHandler AssemblyLoad { add { } remove { } }
-        public event ResolveEventHandler AssemblyResolve { add { } remove { } }
-        public event ResolveEventHandler TypeResolve { add { } remove { } }
-        public event ResolveEventHandler ResourceResolve { add { } remove { } }
-        //public void SetPrincipalPolicy(System.Security.Principal.PrincipalPolicy policy) { }
-        /*
-        public void SetThreadPrincipal(System.Security.Principal.IPrincipal principal)
+        public Assembly[] GetAssemblies() => AssemblyLoadContext.GetLoadedAssemblies();
+        public event AssemblyLoadEventHandler AssemblyLoad 
+        { 
+            add { AssemblyLoadContext.AssemblyLoad += value; } 
+            remove { AssemblyLoadContext.AssemblyLoad -= value; } 
+        }
+        public event ResolveEventHandler TypeResolve
+        { 
+            add { AssemblyLoadContext.TypeResolve += value; } 
+            remove { AssemblyLoadContext.TypeResolve -= value; } 
+        }
+        public event ResolveEventHandler ResourceResolve
+        { 
+            add { AssemblyLoadContext.ResourceResolve += value; } 
+            remove { AssemblyLoadContext.ResourceResolve -= value; } 
+        }
+        public void SetPrincipalPolicy(PrincipalPolicy policy) { }
+        public void SetThreadPrincipal(IPrincipal principal)
```

Comment:
It looks like the thread principal can only be set once, and throws PolicyException on desktop. Do we need to replicate that error case?